sklearn/manifold/_barnes_hut_tsne.pyx

```diff
@@ -202,12 +204,14 @@ cdef void compute_gradient_negative(float[:, :] pos_reference,
         # for the digits dataset, walking the tree
         # is about 10-15x more expensive than the
         # following for loop
-        exponent = (dof + 1.0) / -2.0
+        exponent = (dof + 1.0) / 2.0
```

Comment:
It doesn't look like this has a cdef!! It should also be set at the beginning of the function.

Should probably use cython -a to check for other untyped variables...